misc/gputest.py

```diff
@@ -285,6 +285,46 @@ def _op(self, op):
             self.projects = Util.diff_list(self.projects, del_projects)
         self._log_exec(all_timer.stop(), 'Total %s' % op.capitalize())
 
+    def _update_webgpu_cts_expectations(self, project_root_dir, gpu_device_id):
+        generation = Util.get_intel_gpu_generation('0x%s' % gpu_device_id)
+        if generation == None:
+            Util.warning('Failed to update %s because of unknown generation' % expectation_file)
+            return
+
+        old_gpu_tag = 'intel-gen-9'
+        new_gpu_tag = 'intel-gen-%s' % generation
+        if old_gpu_tag == new_gpu_tag:
+            return
+
+        expectation_file = '%s/%s' % (project_root_dir, 'third_party/dawn/webgpu-cts/expectations.txt')
+        if not os.path.exists(expectation_file):
+            Util.warning('%s does not exist' % expectation_file)
+            return
+
+        tag_header_scope = False
+        new_gpu_tag_existed = False
```

Comment:
nit:
```suggestion
        has_new_gpu_tag = False
```
We should use existing instead of existed for adjective. For convention, we can use has_xxx.